src/Xenial.Tasty.Cli/TastyServer.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Protocols;
+using Xenial.Delicious.Reporters;
+using Xenial.Delicious.Utils;
+
+using static Xenial.Delicious.Utils.Actions;
+
+namespace Xenial.Delicious.Cli
+{
+    public class TastyServer
+    {
+        public event EventHandler<ExecuteCommandEventArgs>? ExecuteCommand;
+        public event EventHandler? CancellationRequested;
+
+        internal async Task DoExecuteCommand(ExecuteCommandEventArgs args)
+        {
+            ExecuteCommand?.Invoke(this, args);
+            await Task.CompletedTask;
+        }
+
+        internal async Task DoRequestCancellation()
+        {
+            CancellationRequested?.Invoke(this, EventArgs.Empty);
+            await Task.CompletedTask;
+        }
+
+        public Task Report(SerializableTestCase @case)
+            => ConsoleReporter.Report(@case);
+
+        public Task Report(IEnumerable<SerializableTestCase> @cases)
+           => ConsoleReporter.ReportSummary(@cases);
+
+        internal Action<IList<SerializableTastyCommand>>? CommandsRegistered;
+        internal Action? EndTestPipelineSignaled;
+        internal Action? TestPipelineCompletedSignaled = () => { };
+        public void RegisterCommands(IList<SerializableTastyCommand> commands)
+            => CommandsRegistered?.Invoke(commands);
+
+        public void SignalEndTestPipeline()
+            => EndTestPipelineSignaled?.Invoke();
+
+        public void SignalTestPipelineCompleted()
+            => TestPipelineCompletedSignaled?.Invoke();
+
+        public void ClearConsole()
```

Comment:
We should at least put some stuff as internal helpers into Tasty it self to avoid the dups here